docs/data/data-grid/pivoting/pivoting.md

```diff
@@ -2,16 +2,138 @@
+In the demo below, the pivot panel is already open and some pivoting parameters have been set.
+Use the **Pivot** switch at the top of the panel to toggle pivoting off and on.
+You can drag and drop existing columns in the **Rows**, **Columns**, and **Values** dropdown menus to change how the data is pivoted.
+
+{{"demo": "GridPivotingMultipleValues.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Pivot model
+
+The pivot model is a configuration object that defines rows, columns, and values of the pivot Grid:
+
+```tsx
+interface GridPivotModel {
+  rows: Array<{
+    field: string;
+    hidden?: boolean;
+  }>;
+  columns: Array<{
+    field: string;
+    sort?: 'asc' | 'desc';
+    hidden?: boolean;
+  }>;
+  values: Array<{
+    field: string;
+    aggFunc: string;
+    hidden?: boolean;
+  }>;
+}
+```
+
+## Initialize pivoting
+
+Use the `initialState` prop to initialize uncontrolled pivoting.
+This is the recommended method unless you specifically need to control the state.
+You can initialize the pivot model, toggle pivot panel visibility, and toggle the pivot mode as shown below:
+
+```tsx
+<DataGridPremium
+  initialState={{ pivoting: { model: pivotModel, active: true, panelOpen: true } }}
+/>
+```
+
+{{"demo": "GridPivotingInitialState.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Controlled pivoting
+
+For fully controlled pivoting state, you can use the following props:
+
+- Pivot model:
+  - `pivotModel`: Controls the current pivot configuration.
+  - `onPivotModelChange`: Callback fired when the pivot model changes.
+- Pivot mode toggle:
+  - `pivotActive`: Controls whether pivot mode is active.
+  - `onPivotActiveChange`: Callback fired when the pivot mode changes between active and inactive.
+- Pivot panel:
+  - `pivotPanelOpen`: Controls whether the pivot panel is open.
+  - `onPivotPanelOpenChange`: Callback fired when the pivot panel is opened or closed.
+
+{{"demo": "GridPivotingControlled.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Using fields in the pivot model multiple times
+
+While this is not supported yet, we are working to bring this feature to the Data Grid.
+
+## Disable pivoting
+
+To disable pivoting, set the `disablePivoting` prop to `true`:
+
+```tsx
+<DataGridPremium disablePivoting />
+```
+
+## Derived columns in pivot mode
+
+In pivot mode, it's often useful to group data by a year or quarter.
+The Data Grid automatically generates year and quarter columns for each **Date** column for this purpose.
+
+For example, the sales dataset we've been using throughout the examples has a **Quarter** column.
+But often in real-world datasets, each sale record would have a precise **Transaction Date** field, like in the following demo.
+
+The **Transaction Date** column is represented by additional columns in pivot mode: **Transaction Date (Year)** and **Transaction Date (Quarter)**:
+
+{{"demo": "GridPivotingFinancial.js", "bg": "inline", "defaultCodeOpen": false}}
```

Comment:
Setting `min` and `max` functions for "Price" and "Volume" in this demo causes `Infinity` and `-∞`:

![image](https://github.com/user-attachments/assets/cd0c3072-958b-413f-8c05-8b30f9d8184a)

Is it intentional? Having empty values would make more sense to me.

Reply:
Good question!
Pivoting highlighted this aggregation behavior in some edge cases, and indeed, it's confusing.

We use infinity as initial values in min/max aggregation functions, but I think they shouldn't ever return these initial values:
https://github.com/mui/mui-x/blob/cef14b8bea193dba5460f379ac3b3e49d671f31b/packages/x-data-grid-premium/src/hooks/features/aggregation/gridAggregationFunctions.ts#L46-L54

Reply:
Done in https://github.com/mui/mui-x/pull/9877/commits/f64494a9dc1e54ce68498042d563fa57c40be19f

Reply:
Looks better now, thanks!